lib/authHttpClient.js

```diff
@@ -0,0 +1,65 @@
+/*!
+ * Copyright (c) 2021 Digital Bazaar, Inc. All rights reserved.
+ */
+import {httpClient} from '@digitalbazaar/http-client';
+import {isInvalidAccessTokenError} from './errorCheck.js';
+import {getAccessToken, getNewAccessToken} from './oauth2-client.js';
+import bedrock from 'bedrock';
+const {BedrockError} = bedrock;
+
+const proxyMethods = new Set([
+  'get', 'post', 'push', 'patch', 'head', 'delete'
+]);
+
+export function getAuthHttpClient({oAuth2Client}) {
```

Comment:
We're creating an instance here and it seems like this is an authorization (vs. authentication) http client so let's throw in the `z` so that's a bit more clear:

```suggestion
export function createAuthzHttpClient({oAuth2Client} = {}) {
```